packages/file_selector/file_selector_windows/lib/src/dart_file_selector_api.dart

```diff
@@ -113,72 +107,110 @@ class DartFileSelectorAPI extends FileDialog {
     String? initialDirectory,
     String? confirmButtonText,
   }) {
+    final SelectionOptions selectionOptions = SelectionOptions(
+        allowMultiple: false, selectFolders: true, allowedTypes: <TypeGroup>[]);
     int hResult = initializeComLibrary();
     final FileOpenDialog dialog = FileOpenDialog.createInstance();
     using((Arena arena) {
       final Pointer<Uint32> options = arena<Uint32>();
       hResult = getOptions(options, hResult, dialog);
-      hResult = setDirectoryOptions(options, hResult, dialog);
+      hResult = setDialogOptions(options, hResult, selectionOptions, dialog);
     });
+
     hResult = addConfirmButtonLabel(dialog, confirmButtonText);
     hResult = _fileOpenDialogAPI.show(hWndOwner, dialog);
-    return returnSelectedElement(hResult, dialog);
+
+    final List<String> selectedPaths =
+        returnSelectedElement(hResult, selectionOptions, dialog);
+    return selectedPaths.isEmpty ? null : selectedPaths.first;
   }
 
   /// Initialices the com library
   @visibleForTesting
   int initializeComLibrary() {
     final int hResult = CoInitializeEx(
         nullptr, COINIT_APARTMENTTHREADED | COINIT_DISABLE_OLE1DDE);
-    if (FAILED(hResult)) {
-      throw WindowsException(hResult);
-    }
+    _validateResult(hResult);
     return hResult;
   }
 
   /// Returns a directory path from user interaction.
   @visibleForTesting
-  String? returnSelectedElement(int hResult, FileOpenDialog dialog) {
-    bool cancelledByUser = false;
-    late String userSelectedPath;
+  List<String> returnSelectedElement(
```

Comment:
```suggestion
  List<String> returnSelectedElements(
```

if this will return a list, we should rename it accordingly